doc/userdoc/contribute/templates_styleguides/styleguide.rst

```diff
@@ -0,0 +1,608 @@
+   :code:`cout << "Hello, World!"`
+
+.. _math_style:
+
+Math equations
+~~~~~~~~~~~~~~
+
+The input language for mathematics is LaTeX markup. See `Mathematics into Type
+<http://www.ams.org/arc/styleguide/mit-2.pdf>`_ for a guide to styling LaTeX math.
+
+
+For equations that take a whole line (or more), use the math directive::
+
+    .. math::
+
+        f(x) = \int_{-\infty}^{\infty} \hat{f}(\xi) e^{2 \pi i x \xi} \, d\xi.
+
+Output rendered as
+
+    .. math::
+
+        f(x) = \int_{-\infty}^{\infty} \hat{f}(\xi) e^{2 \pi i x \xi} \, d\xi.
+
+If the equation runs over several lines you can use ``\\`` as a separator at the end of each line.
+You can also align lines in an equation, using the ``&`` where you want equation aligned::
+
+
+    .. math::
+
+         (a + b)^2  &=  (a + b)(a + b) \\
+                    &=  a^2 + 2ab + b^2
+
+For in-text math, use the math role::
+
+   Now we can see :math:`x=1` for this example.
+
+These will be rendered as
+
+   Now we can see :math:`x=1` for this example.
+
+Admonitions
+~~~~~~~~~~~
+
+Admonitions are directives that render as highlighted blocks to draw the reader's attention to a particular point.
+
+Use them sparingly.
+
+
+Use the admonition
+
+* "See also" to reference internal or external links (only in cases where the reference should stand out),
+
+* "Note" to add additional information that the reader needs to be aware of,
+
+* "Warning" to indicate that something might go wrong without the provided information, and
+
+* "Danger" if the situation may cause severe, possibly irreversible, problems.
+
+
+If you want a custom admonition use::
+
+   .. admonition:: custom label
+
+         Here is some text
+
+Rendered as
+
+
+   .. admonition:: custom label
+
+         Here is some text
+
+
+References
+~~~~~~~~~~
+
+For referencing reStructuredText files within the documentation, use the ``:doc:`` role. It requires the relative path to
+the file::
+
+   :doc:`path/to/file`
+
+In this case, the link caption will be the title of the given document.
+
+You can specify the text you want to use for the link by doing the following::
+
+   :doc:`custom label <path/file>`
+
+This will be rendered as
+
+   :doc:`Top header of file`
+
+   :doc:`custom label <file>`
+
+For cross-referencing specific section headings, figures, or other arbitrary places within file, use the ``:ref:`` role.
+
+The ``:ref:`` role requires a reference label that looks like this ``.. _ref-label:``. Each reference label must be unique
+in the documentation. Separate the reference label from the text it is referecing with a new line::
+
+   .. _my-ref-label:
+
+   Section to cross-reference
+   --------------------------
+
+   Some content in this section.
+   It includes the cross-referecing role :ref:`my-ref-label`.
+
+
+To reference figures or arbitrary places in a file, you must include a custom
+label in the reference for it to work::
+
+    :ref:`custom label <my-arbitrary-place-label>`
```

Comment:
I don not understand the custom label here. Could you give a proper example?

Reply:
added